Fix or suppress javadoc warnings

**Jetty version(s)**
12

**Jetty Environment**
all

**Description**
There are thousands of javadoc warnings that pollute all our builds.
We should either fix them (best option) or suppress them (if we have to).

```

[WARNING] /home/jenkins/agent/workspace/jetty.project_jetty-12.0.x/jetty-core/jetty-http/src/main/java/org/eclipse/jetty/http/BadMessageException.java:24: warning: no comment
[WARNING] public BadMessageException()
[WARNING] ^
[WARNING] /home/jenkins/agent/workspace/jetty.project_jetty-12.0.x/jetty-core/jetty-http/src/main/java/org/eclipse/jetty/http/BadMessageException.java:39: warning: no comment
[WARNING] public BadMessageException(int code)
[WARNING] ^
[WARNING] /home/jenkins/agent/workspace/jetty.project_jetty-12.0.x/jetty-core/jetty-http/src/main/java/org/eclipse/jetty/http/BadMessageException.java:44: warning: no comment
[WARNING] public BadMessageException(int code, String reason)
[WARNING] ^
[WARNING] /home/jenkins/agent/workspace/jetty.project_jetty-12.0.x/jetty-core/jetty-http/src/main/java/org/eclipse/jetty/http/BadMessageException.java:49: warning: no comment
[WARNING] public BadMessageException(int code, String reason, Throwable cause)
[WARNING] ^
[WARNING] /home/jenkins/agent/workspace/jetty.project_jetty-12.0.x/jetty-core/jetty-http/src/main/java/org/eclipse/jetty/http/BadMessageException.java:29: warning: no comment
[WARNING] public BadMessageException(String reason)
[WARNING] ^
[WARNING] /home/jenkins/agent/workspace/jetty.project_jetty-12.0.x/jetty-core/jetty-http/src/main/java/org/eclipse/jetty/http/BadMessageException.java:34: warning: no comment
[WARNING] public BadMessageException(String reason, Throwable cause)
[WARNING] ^
[WARNING] /home/jenkins/agent/workspace/jetty.project_jetty-12.0.x/jetty-core/jetty-http/src/main/java/org/eclipse/jetty/http/ByteRange.java:54: warning: no main description
[WARNING] * @return the length of this byte range
[WARNING] ^
[WARNING] [loading /modules/java.base/java/util/SequencedCollection.class]
[WARNING] /home/jenkins/agent/workspace/jetty.project_jetty-12.0.x/jetty-core/jetty-http/src/main/java/org/eclipse/jetty/http/content/HttpContent.java:89: warning: no main description
[WARNING] * @param path The path within the context to the resource
[WARNING] ^
[WARNING] /home/jenkins/agent/workspace/jetty.project_jetty-12.0.x/jetty-core/jetty-http/src/main/java/org/eclipse/jetty/http/content/CachingHttpContentFactory.java:122: warning: no main description
[WARNING] * @return the max number of cached files.
[WARNING] ^
[WARNING] /home/jenkins/agent/workspace/jetty.project_jetty-12.0.x/jetty-core/jetty-http/src/main/java/org/eclipse/jetty/http/content/CachingHttpContentFactory.java:130: warning: no main description
[WARNING] * @param maxCachedFiles the max number of cached files.
[WARNING] ^
[WARNING] /home/jenkins/agent/workspace/jetty.project_jetty-12.0.x/jetty-core/jetty-http/src/main/java/org/eclipse/jetty/http/content/CachingHttpContentFactory.java:295: warning: no comment
[WARNING] protected class CachedHttpContent extends HttpContent.Wrapper implements CachingHttpContent
[WARNING] ^
[WARNING] /home/jenkins/agent/workspace/jetty.project_jetty-12.0.x/jetty-core/jetty-http/src/main/java/org/eclipse/jetty/http/content/CachingHttpContentFactory.java:282: warning: no comment
[WARNING] protected interface CachingHttpContent extends HttpContent
[WARNING] ^
[WARNING] /home/jenkins/agent/workspace/jetty.project_jetty-12.0.x/jetty-core/jetty-http/src/main/java/org/eclipse/jetty/http/content/CachingHttpContentFactory.java:493: warning: no comment
[WARNING] protected static class NotFoundHttpContent implements CachingHttpContent
[WARNING] ^
[WARNING] /home/jenkins/agent/workspace/jetty.project_jetty-12.0.x/jetty-core/jetty-http/src/main/java/org/eclipse/jetty/http/content/CachingHttpContentFactory.java:78: warning: no comment
[WARNING] public CachingHttpContentFactory(HttpContent.Factory authority, ByteBufferPool bufferPool)
[WARNING] ^
[WARNING] /home/jenkins/agent/workspace/jetty.project_jetty-12.0.x/jetty-core/jetty-http/src/main/java/org/eclipse/jetty/http/content/CachingHttpContentFactory.java:192: warning: no comment
[WARNING] public void flushCache()
```

**How to reproduce?**

Build it

--- a/jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/DateCache.java
+++ b/jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/DateCache.java
@@ -26,7 +26,7 @@ import java.util.TimeZone;
  * <p>
  * If consecutive calls are frequently very different, then this
  * may be a little slower than a normal DateFormat.
- * <p>
+ * </p>
  * @see DateTimeFormatter for date formatting patterns.
  */
 public class DateCache

--- a/jetty-ee10/jetty-ee10-webapp/src/main/java/org/eclipse/jetty/ee10/webapp/WebAppClassLoader.java
+++ b/jetty-ee10/jetty-ee10-webapp/src/main/java/org/eclipse/jetty/ee10/webapp/WebAppClassLoader.java
@@ -124,9 +124,8 @@ public class WebAppClassLoader extends URLClassLoader implements ClassVisibility
      * <p>Run the passed {@link PrivilegedExceptionAction} with the classloader
      * configured so as to allow server classes to be visible</p>
      *
-     * @param <T> The type returned by the action
      * @param action The action to run
-     * @param <T> the type of PrivilegedExceptionAction
+     * @param <T> the type of PrivilegedExceptionAction and the type returned by the action
      * @return The return from the action
      * @throws Exception if thrown by the action
      */

--- a/pom.xml
+++ b/pom.xml
@@ -1409,8 +1409,8 @@
             <charset>UTF-8</charset>
             <docencoding>UTF-8</docencoding>
             <encoding>UTF-8</encoding>
-            <additionalOptions>-html5</additionalOptions>
             <docfilessubdirs>true</docfilessubdirs>
+            <doclint>all,-missing</doclint>
             <detectLinks>false</detectLinks>
             <detectJavaApiLink>false</detectJavaApiLink>
             <detectOfflineLinks>false</detectOfflineLinks>

